compiler/stable_mir/src/mir/body.rs

```diff
@@ -274,6 +274,32 @@ pub enum BinOp {
     Offset,
 }
 
+impl BinOp {
+    /// Return the type of this operation for the given input Ty.
```

Comment:
Should we assert that we're only ever checking this on primitive LHS and RHS so it's not misused?

Reply:
We can, I don't see why not.